### PR TITLE
Tip: Double tilde rounding

### PR DESCRIPTION
**Pull request by pklinger**

### CODE REVIEW

**Comment by drakmail** [comment]
Wow, nice tip

**Comment by loverajoel** [comment]
Wow, do you know something about don't use `~~` in chrome?

**Comment by parro-it** [comment]
Awesome! Maybe it doesn't improve code readability, but it could be useful sometimes.

**Comment by pklinger** [comment]
Yes @loverajoel i've found this here: http://rocha.la/JavaScript-bitwise-operators-in-practice
There are quite some differences amongst the browsers according to the source. Newer browser performance tests even out pretty much... The tip should in fact just save you from one of those "what the fuck does '~~' do??!" moments. I would never ever use this in shared code/projects as ~~ is just not as mnemonic as Math.floor(). As we all know, maintaining code is most expensive, so just avoid it and rather write proper 'understandable' code as long as you don't need some performance gains.

**Comment by gmarketer** [comment]
It should be noted that operation is clipped to 32 bits integers.
So largest number that can be converted that way is 2147483647.

``` javascript
console.log(~~2147483647 === 2147483647) // logs true
console.log(~~(2147483647 + 1) === (2147483647 + 1)) // logs false
```

**Comment by pklinger** [comment]
I should elaborate more on that in another sentence, thanks for the tip. @gmarketer

File: README.md [hunk 1]
```diff
@@ -18,6 +18,32 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter.
...
+console.log(~1337)    // -1338
+
+// numeric input 
+console.log(~~47.11)  // -> 47
+console.log(~~-12.88) // -> -12
+console.log(~~1.9999) // -> 1
+console.log(~~3)      // -> 3
+
+// on failure
+console.log(~~true) // -> 0
```
Comment (hueitan) [inline README.md]: This should be return `1`

File: README.md [hunk 2]
```diff
@@ -18,6 +18,32 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter.
 
 # Tips list
 
+## #XX - Rounding the fast way
+
+> 2016-XX-XX by [pklinger](https://github.com/pklinger)
```
Comment (loverajoel) [inline README.md]: update the date to today

File: README.md [hunk 3]
```diff
@@ -18,6 +18,32 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter.
 
 # Tips list
 
+## #XX - Rounding the fast way
```
Comment (loverajoel) [inline README.md]: update the number to 18

File: README.md (hunk 1 above)
Comment (pklinger) [inline README.md]: thanks for pointing this out. I ll remove it as it could confuse people

**Comment by loverajoel** [comment]
make a merge back, and if you can, please squash your commits at the end

**Comment by pklinger** [comment]
merged. squashing fails constantly :disappointed:

**Comment by loverajoel** [comment]
@pklinger awesome! don't worry about it

**Comment by loverajoel** [comment]
@pklinger hey do you have twitter account?

**Comment by pklinger** [comment]
@loverajoel yes, I do. But I'm more on the 'consumer side' ... https://twitter.com/codebusiness

**Comment by loverajoel** [comment]
@pklinger perfect! Thanks for you awesome tips man!

**Comment by pklinger** [comment]
Everybody just gives a bit and everybody may profit. Thank you for your time you put in here managing it all @loverajoel :clap:

**Comment by julien-f** [comment]
AFAIK, asm.js use `x|0` for that which also cast to a 32 bits integer and consist of only one operation instead of two.

**Pull request merged**